### ISSUE TITLE
How the network should have been defined.

### ISSUE THREAD

**Issue by efleurine** [BroccoliFever/dog_project_udacity#1]
You don't need a dropout in the convo layers because it will introduce randomness while you need to know the exact output of the layer.

```
num_classes = 133
# define the CNN architecture
class Net(nn.Module):
    ### TODO: choose an architecture, and complete the class
    def __init__(self):
        super(Net, self).__init__()
        ## Define layers of a CNN
        self.conv1 = nn.Conv2d(3, 16, 3, padding=1)
        self.conv2 = nn.Conv2d(16, 32, 3, padding=1)
        self.conv3 = nn.Conv2d(32, 64, 3, padding=1)
        self.conv4 = nn.Conv2d(64, 128, 3, padding=1)
        self.pool = nn.MaxPool2d(2, 2)
        self.final_img_dims = 128 x 14 x 14
        # Final image will be downscaled by a factor of 16 from maxpooling
        
        self.fc1 = nn.Linear(self.final_img_dims, 500)
        self.fc2 = nn.Linear(500, num_classes)
        
        self.dropout = nn.Dropout(p=0.4)
    
    def forward(self, x):
        # in =>  x is   3 x 224 x 224
        x = self.pool(F.relu(self.conv1(x)))   # out => x is 16 x 112 x 112
        
        # in x is 16 x 112 x 112
        x = self.pool(F.relu(self.conv2(x))) # out x is 32 x 56 x 56
        
        # in x is 32 x 56 x 56
        x = self.pool(F.relu(self.conv3(x)))  # out  x is 64 x 28 x 28
       
       # in x is 64 x 28 x 28
       x = self.pool(F.relu(self.conv4(x))) #  out  x is 128 x 14 x 14

        # Fully Connected Layer #
        x = x.view(-1, 128 * 14 * 14)

        x =  F.relu(self.fc1(self.dropout(x)))
        x =  self.fc2(self.dropout(x))
 
        return x
```

**Comment by gregor-io** [comment]
Done, added batch normalization instead
